### ISSUE TITLE
Personal opinion about readme

### ISSUE THREAD

**Issue by advename** [iconfu/svg-inject#17]
I like this small library, but would maybe suggest to use a different approach as an example.

**Example:**

```html
<html>
<head>

</head>
<body>
  <img src="image.svg"  class="inject-svg"/>

  <script src="svg-inject.min.js"></script>
  <script>
   let svg = document.querySelector(".inject-svg");

   document.addEventListener("DOMContentLoaded",init);

   function init(){
      SVGInject(svg);
   }
</script>
</body>
</html>
```

<hr>

The reason behind this is, of course, it's not as easy and short as the other one, but if you have other scripts running beside it and are only using the previous method, you'll get stuck because you might then have multiple onload's running and the SVG it hasn't been injected yet for the "main.js" script.

**Comment by waruyama** [comment]
Thanks for the input.

With SVGInject you can use both approaches, either using the `onload` attribute or injecting the SVGs manually as in your example. The advantage of using `onload` is that some images can be injected while the website is still loading and if new `<img>` elements are added dynamically injection will occur automatically. Therefore we prefer using the `onload` attribute. we have updated the documentation on the [development branch](https://github.com/iconfu/svg-inject/tree/development) if you want to take a look.

Note that in either case the SVGs will be loaded asynchronously by the injector. This means that in your example the injection may be finished before other scripts are executed, but there is no guarantee that this is the case.

If you want to make sure that the SVGs have been injected before you run any other scripts you will have to use the `afterInject` hook to do so. Here is how this can be done in combination with your example:

    SVGInject.setOptions({
      afterInject: function(img, svg) {
        if (document.querySelector("img.inject-svg") == null) {
          // All SVGs have been injected, we can start further processing
        }
      },
      onFail: function(img) {
        img.classList.remove('inject-me');
      }
    });

**Comment by waruyama** [comment]
With the [changes on the development branch](https://github.com/iconfu/svg-inject/tree/development#basic-usage), do you think this issue can be closed?

**Comment by advename** [comment]
Yes, with the single difference that the script is set in the bottom. If an example is displayed it should be done the non-render-blocking way.

Thanks!
